src/SelectableBox/index.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Form from '../Form';
+import SelectableBoxSet from './SelectableBoxSet';
+import { useCheckboxSetContext } from '../Form/FormCheckboxSetContext';
+import { useRadioSetContext } from '../Form/FormRadioSetContext';
+
+const INPUT_TYPES = [
+  'radio',
+  'checkbox',
+];
+
+const SelectableBox = ({
+  type,
+  value,
+  checked,
+  children,
+  isIndeterminate,
+  isInvalid,
+  onClick,
+  onFocus,
+  inputHidden,
+  className,
+}) => {
+  const getInputType = () => {
+    switch (type) {
+      case 'radio':
+        return Form.Radio;
+      case 'checkbox':
+        return Form.Checkbox;
+      default:
+        return Form.Radio;
+    }
+  };
+
+  const isChecked = () => {
+    switch (type) {
+      case 'radio':
+        return useRadioSetContext().value === value;
+      case 'checkbox':
+        return useCheckboxSetContext().value?.includes(value);
+      default:
+        return useRadioSetContext().value === value;
+    }
+  };
+
+  const inputRef = useRef(null);
+  const input = React.createElement(getInputType(), {
+    value,
+    checked,
+    hidden: inputHidden,
+    ref: inputRef,
+    tabIndex: -1,
+    onChange: () => {},
+    ...(type === 'checkbox' && { isIndeterminate }),
+  }, null);
+
+  const onClickHandler = () => {
+    inputRef.current.click();
+    if (onClick) { onClick(inputRef.current); }
+  };
+
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
```

Comment:
Hmm, what would it take to not need to disable this a11y warning?

Reply:
It requires a key event for the `SelectableBox`. Seems like adding `onKeyPress={onClickHandler}` resolves this warning. Now `Tab` + `Enter` allows to select/deselect

Reply:
**notice:** for current implementation if user focuses the `SelectableBox` `onClickHandler` is triggered by any key. Should I reduce those keys only to 'Enter'?